### PR TITLE
172722 - get upns from shop facade stub

### PR DESCRIPTION
**Pull request by chaital14503**
This PBI consists changes of [PBI 172722](https://dev.azure.com/ukhydro/S-100%20Permit%20Generator/_workitems/edit/172722)

Code changes added
Unit tests added
Warnings suppressed - 8603, 8604, 8625 
Functional Tests added

### CODE REVIEW

File: tests/UKHO.S100PermitService.Common.UnitTests/Helpers/UserPermitApiClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        }
+
+        [Test]
+        [TestCase(HttpStatusCode.BadRequest)]
+        [TestCase(HttpStatusCode.NotFound)]
+        [TestCase(HttpStatusCode.Unauthorized)]
+        [TestCase(HttpStatusCode.InternalServerError)]
+        [TestCase(HttpStatusCode.ServiceUnavailable)]
+        [TestCase(HttpStatusCode.UnsupportedMediaType)]
+        public void WhenUserPermitServiceResponseOtherThanOk(HttpStatusCode httpStatusCode)
```
Comment (shirinbt14926) [inline tests/UKHO.S100PermitService.Common.UnitTests/Helpers/UserPermitApiClientTests.cs:55]:
```suggestion
        public void WhenUserPermitServiceResponseOtherThanOk_ThenResponseShouldNotBeOk(HttpStatusCode httpStatusCode)
```

**Review by shirinbt14926 — COMMENTED** [review]
Minor comments please

File: tests/UKHO.S100PermitService.Common.UnitTests/Helpers/UserPermitApiClientTests.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        private const string ValidResponse = "[{\r\n  \"licenceId\": \"1\",\r\n  \"userPermits\": [\r\n    {\r\n      \"title\": \"Port Radar\",\r\n      \"upn\": \"FE5A853DEF9E83C9FFEF5AA001478103DB74C038A1B2C3\"\r\n    }\r\n  ]\r\n}]";
+
+        [SetUp]
+        public void SetUp()
+        {
+            _fakeHttpClientFactory = A.Fake<IHttpClientFactory>();
+        }
+
+        [Test]
+        public void WhenValidDataIsPassed_ThenUserPermitServiceReturnsOKResponse()
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply tests/UKHO.S100PermitService.Common.UnitTests/Helpers/UserPermitApiClientTests.cs:26]: Done

File: tests/UKHO.S100PermitService.Common.UnitTests/Helpers/UserPermitApiClientTests.cs (hunk 1 above)
Comment (chaital14503) [reply tests/UKHO.S100PermitService.Common.UnitTests/Helpers/UserPermitApiClientTests.cs:55]: Done

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Review by Mayuresh-Gawde — APPROVED** [review]
LGTM from QA!!

File: src/UKHO.S100PermitService.API/Program.cs [hunk 3]
```diff
@@ -112,6 +112,7 @@ private static void ConfigureServices(WebApplicationBuilder builder)
 
             builder.Services.Configure<EventHubLoggingConfiguration>(builder.Configuration.GetSection("EventHubLoggingConfiguration"));
             builder.Services.Configure<HoldingsServiceApiConfiguration>(configuration.GetSection("HoldingsServiceApiConfiguration"));
+            builder.Services.Configure<UserPermitServiceApiConfiguration>(configuration.GetSection("UserPermitServiceApiConfiguration"));
```
Comment (jrippington) [inline src/UKHO.S100PermitService.API/Program.cs:115]: Put the section names into const string

File: src/UKHO.S100PermitService.API/Program.cs [hunk 4]
```diff
@@ -141,15 +142,25 @@ private static void ConfigureServices(WebApplicationBuilder builder)
                 client.Timeout = TimeSpan.FromMinutes(holdingsServiceApiConfiguration.RequestTimeoutInMinutes);
             });
 
+            var userPermitServiceApiConfiguration = builder.Configuration.GetSection("UserPermitServiceApiConfiguration").Get<UserPermitServiceApiConfiguration>();
```
Comment (jrippington) [inline src/UKHO.S100PermitService.API/Program.cs:145]: Put the section names into const string so that if they are changed, all instances are updated

**Review by jrippington — CHANGES_REQUESTED** [review]
Very minor tidy up of config section names, thanks

**Review by shirinbt14926 — APPROVED** [review]
LGTM. Thanks for making changes @chaital14503

**Review by shirinbt14926 — APPROVED** [review]

File: src/UKHO.S100PermitService.API/Program.cs (hunk 3 above)
Comment (chaital14503) [reply src/UKHO.S100PermitService.API/Program.cs]: Done

File: src/UKHO.S100PermitService.API/Program.cs (hunk 4 above)
Comment (chaital14503) [reply src/UKHO.S100PermitService.API/Program.cs]: Done

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

**Pull request merged**